collator/src/lib.rs

```diff
@@ -359,7 +359,7 @@ fn build_collator_service<SP, P, C, R, Extrinsic>(
 
 			let future = silenced.map(drop);
 
-			tokio::spawn(future);
+			spawner.spawn(future).expect("could not run build collator task");
```

Comment:
This panicker is not allowed. The error should be handled in a more robust way.

Reply:
It won't be anymore robust as it is a SpawnTaskHandle behind. I replaced the generic by the actual type and removed the exception in a88d32fe4bafda1e6ac11bd532597884468e3649

Reply:
I can add a SpawnEssentialTaskHandle to the function if you want. With that it will stop the whole node if an error occurs

Reply:
OK, it seems better now but I do want to clarify:
  - Panics are not allowed unless they can be proven to be impossible
  - It isn't reasonable to rely on cross-module concerns to prove a panic responsible. so assuming that the generic type is some specific type in practice, or assuming that a function is called only a certain way is not OK

The new code you pushed does not have these problems, but I hope it is helpful to explain what I was referring to with my first comment.

Reply:
Sure, I will try to panic less :scream: